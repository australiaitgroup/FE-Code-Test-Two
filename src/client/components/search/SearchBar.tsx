import * as React from 'react';
import { Button, Input } from 'semantic-ui-react';

import { ISearchBarProps } from '../interfaces/searchBar.interface';

let searchTimer;

const SearchBar: React.FunctionComponent<ISearchBarProps> = ({
    handleInputChange,
    search,
}) => (
    <Input
        action={
            <Button
                color="teal"
                content="Search"
                onClick={search}
            />
        }
        fluid
        icon="search"
        iconPosition="left"
        onKeyUp={(event) => {
            clearTimeout(searchTimer);
            handleInputChange(event.target.value);
            if (event.key === 'Enter') return search();
            searchTimer = setTimeout(search, 300);
        }}
        placeholder="Search..."
    />
);

export default SearchBar;
