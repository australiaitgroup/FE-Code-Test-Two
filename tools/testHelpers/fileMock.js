// Used by Jest: refer to https://jestjs.io/docs/en/webpack.html

const path = require('path');

module.exports = {
    process(src, filename) {
        return `module.exports = ${JSON.stringify(path.basename(filename))};`;
    },
};
