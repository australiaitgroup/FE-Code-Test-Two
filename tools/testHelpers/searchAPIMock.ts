import * as nock from 'nock';

export default nock('https://hn.algolia.com/api/v1');
