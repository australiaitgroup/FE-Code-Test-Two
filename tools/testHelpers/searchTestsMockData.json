{
  "searchResult": {
    "hits": [
      {
        "created_at": "2015-06-03T19:45:33.000Z",
        "title": null,
        "url": null,
        "author": "pingec",
        "points": null,
        "story_text": null,
        "comment_text": "test",
        "num_comments": null,
        "story_id": 9654758,
        "story_title": "How Well Can You Hear Audio Quality?",
        "story_url": "http://www.npr.org/sections",
        "parent_id": 9654758,
        "created_at_i": 1433360733,
        "objectID": 32223433,
        "_highlightResult": {
          "author": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "pingec"
          },
          "comment_text": {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": ["testttd"],
            "value": "Similar test: <a href='http://test.tidalhifi.com'"
          },
          "story_title": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "How Well Can You Hear Audio Quality?"
          },
          "story_url": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "http://www.npr.org/sections/therecord/2015/06/02/"
          }
        },
        "_tags": ["comment", "author_pingec", "story_9654758"]
      }
    ],
    "nbHits": 1,
    "page": 0,
    "nbPages": 1,
    "hitsPerPage": 20,
    "processingTimeMS": 10,
    "exhaustiveNbHits": true,
    "query": "test",
    "params": "advancedSyntax=true&analytics=false&hitsPerPage=20&query=testttd&tags=%28story%2Ccomment%29"
  }
}
