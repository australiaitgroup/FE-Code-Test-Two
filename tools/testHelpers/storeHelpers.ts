import { Store } from 'redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import reducers from '../../src/client/reducers';
import { IState } from '../../src/types/stateAndAction';

export function getInitialState(): IState {
    return reducers(undefined, { type: null });
}

export function createMockStore(initialState: IState): Store {
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    return mockStore(initialState);
}

export function deepClone(toBeCloned: any): any {
    return JSON.parse(JSON.stringify(toBeCloned));
}
