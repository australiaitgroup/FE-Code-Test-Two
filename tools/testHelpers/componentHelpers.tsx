import * as React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import * as renderer from 'react-test-renderer';

import { createMockStore } from './storeHelpers';

export function renderComponent(children, storeData) {
    const store = createMockStore(storeData);

    return renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>);
}
