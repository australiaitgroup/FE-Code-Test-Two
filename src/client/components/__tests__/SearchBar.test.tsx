import * as React from 'react';

import SearchBar from '../search/SearchBar';
import { renderComponent } from '../../../../tools/testHelpers/componentHelpers';

it('SearchBar renders correctly', () => {
    const props = {
        handleInputChange: () => {},
        search: () => Promise.resolve(),
    };

    const component = renderComponent(<SearchBar {...props} />, {});
    const result = component.toJSON();
    expect(result).toMatchSnapshot();
});
