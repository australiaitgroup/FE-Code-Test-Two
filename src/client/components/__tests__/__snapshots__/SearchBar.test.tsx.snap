// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="ui fluid action left icon input"
>
  <input
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search..."
    type="text"
  />
  <i
    aria-hidden="true"
    className="search icon"
  />
  <button
    className="ui teal button"
    onClick={[Function]}
    role="button"
  >
    Search
  </button>
</div>
`;
